src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -4415,24 +4441,32 @@ def update_sku(self, mc: ManagedCluster) -> ManagedCluster:
         """
         self._ensure_mc(mc)
 
+        # there are existing MCs with nil sku, that is Base/Free
+        if mc.sku is None:
+            mc.sku = self.models.ManagedClusterSKU(
+                sku="Base",
+                tier="Free",
+            )
+        skuName = self.context.get_sku_name()
+
+        if skuName is not None and skuName == CONST_MANAGED_CLUSTER_SKU_NAME_AUTOMATIC:
+            mc.sku.name = "Automatic"
+            # passive Kind should always to match sku.name
+            mc.kind = "Automatic"
+        else:
+            mc.sku.name = "Base"
+            # passive Kind should always match sku.name
+            mc.kind = "Base"
+
         # Premium without LTS is ok (not vice versa)
         if self.context.get_tier() == CONST_MANAGED_CLUSTER_SKU_TIER_PREMIUM:
```

Comment:
Please fix failed [checks](https://dev.azure.com/azclitools/public/_build/results?buildId=146939&view=logs&j=59e8686e-5e74-514d-6cad-f7d66c66b425&t=20180a09-30e4-5d69-fc0c-4652edaa2d05&l=2105)

> FAILED src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py::AKSPreviewManagedClusterCreateDecoratorTestCase::test_construct_mc_profile_preview
> FAILED src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py::AKSPreviewManagedClusterUpdateDecoratorTestCase::test_update_mc_profile_preview

With no user input, previously the sku profile is left null but after your change, sku.name is set to Base

Reply:
fixed and passed all UTs check